src/sentry/lang/native/applecrashreport.py

```diff
@@ -173,8 +173,8 @@ def _convert_debug_meta_to_binary_image_row(self, debug_image):
         return "%s - %s %s %s  <%s> %s" % (
             hex(image_addr),
             hex(image_addr + debug_image["image_size"] - 1),
-            image_name(debug_image["name"]),
+            image_name(debug_image.get("name", NATIVE_UNKNOWN_STRING)),
             self.context["device"]["arch"],
             (debug_image.get("id") or debug_image.get("uuid")).replace("-", "").lower(),
-            debug_image["name"],
+            debug_image.get("name", NATIVE_UNKNOWN_STRING),
```

Comment:
Wait, `"name"` is long gone and now normalized into `"code_file"` which should be used here, instead. I'm afraid tests for this are very outdated :(

Reply:
The schema is documented here: 
https://docs.sentry.io/development/sdk-dev/event-payloads/debugmeta/

Schema implementation is here:
https://github.com/getsentry/semaphore/blob/2e39b6dd6cf64a15d13800f68af2f86d0a8bcaf9/general/src/protocol/debugmeta.rs#L125

Normalization of legacy types is implemented here:
https://github.com/getsentry/semaphore/blob/2e39b6dd6cf64a15d13800f68af2f86d0a8bcaf9/general/src/store/legacy.rs#L19

Reply:
Also `name` is required according to https://github.com/getsentry/semaphore/blob/2e39b6dd6cf64a15d13800f68af2f86d0a8bcaf9/general/src/protocol/debugmeta.rs#L38

it's just docs being outdated

Reply:
Not quite, `AppleDebugImage` is deprecated and normalized into `NativeDebugImage`, which only has a `"code_file"`. Docs are definitely up to date.

Reply:
seems like semaphore says the name is required

Reply:
(ignore last comment)

Reply:
Here is a somewhat realistic snapshot of a current debug image payload, unfortunately in yaml this time:

https://github.com/getsentry/sentry/blob/e0d23220f81c4a01c66b8a8298e952dfca8ffbb9/tests/symbolicator/snapshots/SymbolicatorMinidumpIntegrationTest/test_full_minidump.pysnap#L16-L30

Reply:
:+1: Thanks! I'll also try to update these tests to be closer here too.

Reply:
@untitaker @jan-auer  Actually before I keep making changes.
is `code_file` optional? I think I might be misunderstanding the process here.

I see required here: https://github.com/getsentry/semaphore/blob/2e39b6dd6cf64a15d13800f68af2f86d0a8bcaf9/general/src/protocol/debugmeta.rs#L132

But optional here: https://docs.sentry.io/development/sdk-dev/event-payloads/debugmeta/#debug-images

Is it optional from the user side, but during processing it becomes required (and defaulted I assume)?